context.go

```diff
@@ -0,0 +1,39 @@
+package rpc
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+)
+
+// CtxRpcKey is a type defining the context key for the RPC context
+type CtxRpcKey int
+
+const (
+	// CtxRpcTagsKey defines a context key that can hold a slice of context keys
+	CtxRpcTagsKey CtxRpcKey = iota
+)
+
+type CtxRpcTags map[string]interface{}
+
+// NewContext returns a new Context that carries adds the given log
+// tag mappings (context key -> display string).
+func NewContextWithLogTags(ctx context.Context, logTagsToAdd CtxRpcTags) (context.Context, error) {
+	currTags, ok := RpcTagsFromContext(ctx)
+	if !ok {
```

Comment:
I was using the existing pattern from the standard logger in client. I don't have an opinion on any of the questions you've asked, so if you feel more strongly either way, I'll happily change it.